### PR TITLE
Cut endpoint implementation (retrieve)

### LINKED ISSUE #218: Query the current cut from a Chainweb node
ℹ️ This issue is part of Epic #201

## Objective

Implement the GET request: [Query the current cut from a Chainweb node](https://api.chainweb.com/openapi/#tag/cut/paths/~1cut/get)

### PR DESCRIPTION
**Pull request by EdwinGuayacan**
Fixes #218 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [x] This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: lib/chainweb/p2p/cut.ex [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  @type opts :: Keyword.t()
+  @type error :: {:error, Error.t()}
+  @type cut_response :: CutResponse.t() | error()
+
+  @spec retrieve(opts :: opts()) :: cut_response()
+  def retrieve(opts) do
+    location = Keyword.get(opts, :location)
+    network_id = Keyword.get(opts, :network_id, :testnet04)
+    query_params = Keyword.get(opts, :query_params, nil)
```
Comment (miguelnietoa) [inline lib/chainweb/p2p/cut.ex]: @EdwinGuayacan some suggestions here!

1. We should set a default value for `opts`, it means `opts \\ []`.
So to make a request to `https://api.chainweb.com/chainweb/0.0/mainnet01/cut` the user only had to call:
    ```elixir
    Kadena.Chainweb.P2P.Cut.retrieve()
    ```

2. In line 19, use `Keyword.get/2` instead of sending `nil` as the third arg. https://hexdocs.pm/elixir/1.12/Keyword.html#get/3
3. IMO, a more descriptive approach to query params is:
    ```elixir
    Kadena.Chainweb.P2P.Cut.retrieve(query_params: [maxheight: 123456])
    ```
    In this way, it's clear we're setting a max height.

File: lib/chainweb/p2p/cut_response.ex [hunk 2]
```diff
@@ -0,0 +1,23 @@
+defmodule Kadena.Chainweb.P2P.CutResponse do
+  @moduledoc """
+  `CutResponse` struct definition.
+  """
+  @behaviour Kadena.Chainweb.Type
+
+  @type hashes :: map()
+  @type weight :: String.t()
+  @type height :: integer()
```
Comment (miguelnietoa) [inline lib/chainweb/p2p/cut_response.ex]:
```suggestion
  @type height :: non_neg_integer()()
```

File: lib/chainweb/request.ex [hunk 3]
```diff
@@ -49,8 +51,9 @@ defmodule Kadena.Chainweb.Request do
     :segment_path,
     :query,
     :headers,
-    :body,
-    :encoded_query
+    :encoded_query,
+    :location,
+    body: ""
```
Comment (miguelnietoa) [inline lib/chainweb/request.ex:56]: @EdwinGuayacan why is this required?

**Review by miguelnietoa — CHANGES_REQUESTED** [review]
@EdwinGuayacan @Odraxs really good job! 👏🏻 

Missing tests for CutResponse type. More comments added

File: lib/chainweb/request.ex (hunk 3 above)
Comment (EdwinGuayacan) [reply lib/chainweb/request.ex:56]: because in the `request/5` function of the `ClientDefault` it doesn't allow nil body:
image

Comment (miguelnietoa) [reply lib/chainweb/request.ex:56]: Got it, thanks!

**Review by miguelnietoa — APPROVED** [review]
